#! /usr/bin/env python
# Forearm camera configuration

PACKAGE='buoy_finder'
import roslib; roslib.load_manifest(PACKAGE)

from math import pi

from driver_base.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()
#Name            Type   Reconfiguration level             Description         Default Min Max
gen.add("h_min", int_t, SensorLevels.RECONFIGURE_RUNNING,"Minimum Hue Level.",75, 0, 255)
gen.add("h_max", int_t, SensorLevels.RECONFIGURE_RUNNING,"Maximum Hue Level.",220,0, 255)
gen.add("s_min", int_t, SensorLevels.RECONFIGURE_RUNNING,"Minimum Saturation Level.",0, 0, 255)
gen.add("s_max", int_t, SensorLevels.RECONFIGURE_RUNNING,"Maximum Saturation Level.",255, 0, 255)
gen.add("mode",  int_t, SensorLevels.RECONFIGURE_RUNNING,"cvFilterHS mode",0,0,1)

exit(gen.generate(PACKAGE, "dynamic_reconfigure_node", "BuoyFinder"))
